File: 2026-08/src/day_29/prms_43162/Solution.py
from collections import deque


def solution(n, computers):
    vis = [False] * n

    def bfs(start):
        q = deque([start])
        vis[start] = True
        while q:
            cur = q.popleft()
            for nxt in range(n):
                if computers[cur][nxt] == 0 or vis[nxt]:
                    continue
                q.append(nxt)
                vis[nxt] = True

    cnt = 0
    for i in range(n):
        if vis[i]:
            continue
        bfs(i)
        cnt += 1

    return cnt

File: 2026-08/src/day_29/prms_43162/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

const int MAXN = 200;
bool vis[MAXN];

void bfs(int start, int n, vector<vector<int>>& computers) {
    queue<int> q;
    q.push(start);

    vis[start] = true;

    while (!q.empty()) {
        int cur = q.front();
        q.pop();

        for (int nxt = 0; nxt < n; nxt++) {
            if (computers[cur][nxt] == 0 || vis[nxt]) continue;
            q.push(nxt);
            vis[nxt] = true;
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int cnt = 0;

    for (int i = 0; i < n; i++) {
        if (vis[i]) continue;
        bfs(i, n, computers);
        cnt++;
    }

    return cnt;
}
